Content.Shared/Mech/Components/MechComponent.cs

```diff
@@ -92,6 +92,9 @@ public sealed partial class MechComponent : Component
     [ViewVariables, AutoNetworkedField]
     public EntityUid? CurrentSelectedEquipment;
 
+    [DataField]
+    public bool AffectedByEmp = true;
```

Comment:
```suggestion
    public bool IsEmpVulnerable = true;
```

Это имя подходит больше и лучше объясняет что делает этот бул